bpf/lib/conntrack.h

```diff
@@ -1076,10 +1085,19 @@ ct_has_nodeport_egress_entry6(const void *map,
 	entry = map_lookup_elem(map, ingress_tuple);
 	ingress_tuple->flags = prev_flags;
 
-	if (entry)
-		return entry->node_port || (check_dsr && entry->dsr);
+	if (!entry)
+		return false;
 
-	return 0;
+	if (entry->node_port) {
+		if (rev_nat_index)
+			*rev_nat_index = entry->rev_nat_index;
+		return true;
+	}
+
+	if (check_dsr && entry->dsr)
+		return true;
+
+	return false;
```

Comment:
The code is identical with the v4 case. Maybe you can wrap the error checking and saving the `rev_nat_index` in an inline function?

Reply:
I like it :+1: See the pushed diff - two birds, one stone.